optimum_benchmark/scenarios/energy_star/preprocessing_utils.py

```diff
@@ -45,3 +71,407 @@ def tokenize_function(examples):
     )  # We don't want a torch dependency here but for now the only backend used for this benchmark is PyTorch
 
     return dataset
+
+
+def summarization_preprocessing(
+    dataset: Dataset,
+    config: EnergyStarConfig,
+    tokenizer: PreTrainedTokenizer,
+    pretrained_config: PretrainedConfig,
+) -> Dataset:
+    # Remove empty samples when batch_size is 1 because empty inputs will make the model fail
+    if config.input_shapes["batch_size"] == 1:
+        dataset = dataset.filter(lambda example: example[config.text_column_name] != "")
+
+    if config.num_samples != -1:
+        dataset = dataset.select(range(config.num_samples))
+
+    # Add a pad token if the tokenizer doesn't have one
+    if getattr(tokenizer, "pad_token", None) is None:
+        tokenizer.pad_token = tokenizer.eos_token
+
+    padding = False if config.input_shapes["batch_size"] == 1 else True
+
+    def tokenize_function(examples):
+        return tokenizer(
+            examples[config.text_column_name],
+            padding=padding,
+            truncation=config.truncation,
+            max_length=getattr(pretrained_config, "max_position_embeddings", 512),
+        )
+
+    dataset = dataset.map(
+        tokenize_function,
+        batched=True,
+        remove_columns=dataset.features,
+        desc="Running tokenizer on dataset",
+    ).with_format("torch")
+
+    return dataset
+
+
+def text_classification_preprocessing(
+    dataset: Dataset,
+    config: EnergyStarConfig,
+    tokenizer: PreTrainedTokenizer,
+    pretrained_config: PretrainedConfig,
+) -> Dataset:
+    # Remove empty samples when batch_size is 1 because empty inputs will make the model fail
+    if config.input_shapes["batch_size"] == 1:
+        dataset = dataset.filter(lambda example: example[config.text_column_name] != "")
+
+    if config.num_samples != -1:
+        dataset = dataset.select(range(config.num_samples))
+
+    # Add a pad token if the tokenizer doesn't have one
+    if getattr(tokenizer, "pad_token", None) is None:
+        tokenizer.pad_token = tokenizer.eos_token
```

Comment:
repeated multiple times, makes more sense to do this after tokenizer instantiation (in `transformers_utils.py`)